### ISSUE TITLE
Optional of subclass of unimported baseclass immediately reduces to None

### ISSUE THREAD

**Issue by rkr-at-dbx** [python/mypy#2570]
With `--strict-optional` and `--silent-imports`:

```
from typing import Optional
from unchecked_module import BaseClass

class Subclass(BaseClass):
	pass

_instance = None  # type: Optional[Subclass]
reveal_type(_instance)  # error: Revealed type is 'builtins.None'
```

This remains true even if the assignment is changed to `_instance = Subclass()`.

**Comment by gvanrossum** [comment]
I stumbled upon this and I can't repro it any more. Ray, can you? If not let's close it, it's probably been fixed by one of the many changes in the last 5 months.

**Comment by rkr-at-dbx** [comment]
Bisection confirms that this was fixed by commit 9496d6fff275162e067b89010d509fc618d8ec38. Closing.
